components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m

```diff
@@ -87,4 +87,19 @@ - (void)testDraggedRippleView {
   [self generateSnapshotAndVerifyForView:self.view];
 }
 
+- (void)testRippleHighlightedIsNotDissolvedWhenSetToYES {
+  // Given
+  self.rippleView.layer.speed = 10000;
+  [self.rippleView setRippleColor:UIColor.orangeColor forState:MDCRippleStateHighlighted];
+  self.rippleView.rippleHighlighted = YES;
+  [self.rippleView setValue:@YES forKey:@"_tapWentInsideOfBounds"];
+  [self.rippleView setValue:@YES forKey:@"_tapWentOutsideOfBounds"];
+  self.rippleView.rippleHighlighted = NO;
+  [self.rippleView setValue:@NO forKey:@"_tapWentOutsideOfBounds"];
+  self.rippleView.rippleHighlighted = YES;
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.view];
```

Comment:
Will the animations have had a chance to run with the way this test is written? Core Animation animations are typically added at the turn of the run loop but I don't see a moment where we let the run loop run here.

Reply:
That's why I set the speed to 2000 for the ripple layer to allow the animations to run instantly.

Reply:
added a runloop run instead of a layer speed change.

Reply:
To clarify: the speed won't take any effect until the run loop has had a chance to commit the animations to the render server, so a run loop pump (or a `[CATransaction flush]`) is necessary for the animations to take effect.

Reply:
Done! thanks